src/DispatchedTuples.jl

```diff
@@ -209,7 +209,7 @@ show_default(io::IO, dt::DispatchedTuple, ::NoDefaults) = println(io, "  default
 show_default(io::IO, dt::DispatchedSet, ::NoDefaults) = println(io, "  default => error")
 
 function Base.show(io::IO, dt::AbstractDispatchedTuple)
-    show(io, typeof(dt))
+    print(io, "$(nameof(typeof(dt)))")
```

Comment:
I wonder if `Base.typename` is what you really want? Maybe they're equivalent though?

```suggestion
    print(io, "$(Base.typename(dt))")
```

Reply:
I haven't seen `Base.typename` before, I'm not exactly sure what it does
```julia
julia> struct Foo{A,B}
       a::A
       b::B
       end

julia> Base.typename(Foo(1,2))
ERROR: typename does not apply to this type
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] typename(a::Foo{Int64, Int64})
   @ Base ./essentials.jl:303
 [3] top-level scope
   @ REPL[2]:1

julia> Base.typename(typeof(Foo(1, "2")))
typename(Foo)
```

Reply:
I'll stick with `nameof(typeof())` for now.

Reply:
Ah sorry my bad. I thought `Base.typename(typeof(Foo(1, "2")))` would return `Foo` but not sure how I used it now...